QuantConnect.ExanteBrokerage/ExanteBrokerage.cs

```diff
@@ -0,0 +1,845 @@
+            return !symbol.Value.Contains("-UNIVERSE-");
+        }
+
+        /// <summary>
+        /// Create a tick from the Exante feed stream data
+        /// </summary>
+        /// <param name="exanteFeedTrade">Exante feed stream data object</param>
+        /// <returns>LEAN Tick object</returns>
+        private Tick CreateTick(ExanteFeedTrade exanteFeedTrade)
+        {
+            var symbolId = exanteFeedTrade.SymbolId;
+            if (!_subscribedTickers.TryGetValue(symbolId, out var symbol))
+            {
+                return null; // Not subscribed to this symbol.
+            }
+
+            if (exanteFeedTrade.Size == decimal.Zero)
+            {
+                return null;
+            }
+
+            // Convert the timestamp to exchange timezone and pass into algorithm
+            var time = GetRealTimeTickTime(exanteFeedTrade.Date, symbol);
+
+            var instrument = Client.GetSymbol(symbolId);
+
+            var size = exanteFeedTrade.Size ?? 0m;
+            var price = exanteFeedTrade.Price ?? 0m;
+            var tick = new Tick(time, symbol, "", instrument.Exchange, size, price);
+            return tick;
+        }
+
+        /// <summary>
+        /// Returns a timestamp for a tick converted to the exchange time zone
+        /// </summary>
+        private DateTime GetRealTimeTickTime(DateTime time, Symbol symbol)
+        {
+            DateTimeZone exchangeTimeZone;
+            if (!_symbolExchangeTimeZones.TryGetValue(symbol, out exchangeTimeZone))
+            {
+                // read the exchange time zone from market-hours-database
+                exchangeTimeZone = MarketHoursDatabase.FromDataFolder()
+                    .GetExchangeHours(symbol.ID.Market, symbol, symbol.SecurityType).TimeZone;
+                _symbolExchangeTimeZones.Add(symbol, exchangeTimeZone);
+            }
+
+            return time.ConvertFromUtc(exchangeTimeZone);
+        }
+
+        /// <summary>
+        /// Create a tick from the Exante tick shorts stream data
+        /// </summary>
+        /// <param name="exanteTickShort">Exante tick short stream data object</param>
+        /// <returns>LEAN Tick object</returns>
+        private Tick CreateTick(ExanteTickShort exanteTickShort)
+        {
+            if (!_subscribedTickers.TryGetValue(exanteTickShort.SymbolId, out var symbol))
+            {
+                // Not subscribed to this symbol.
+                return null;
+            }
+
+            var time = GetRealTimeTickTime(exanteTickShort.Date, symbol);
+            var bids = exanteTickShort.Bid.ToList();
+            var asks = exanteTickShort.Ask.ToList();
+            return new Tick(time, symbol, "", "",
+                bids.IsNullOrEmpty() ? decimal.Zero : bids[0].Size,
+                bids.IsNullOrEmpty() ? decimal.Zero : bids[0].Price,
+                asks.IsNullOrEmpty() ? decimal.Zero : asks[0].Size,
+                asks.IsNullOrEmpty() ? decimal.Zero : asks[0].Price);
```

Comment:
My concern is `bids` and `asks` are a list and you are only using the first entry and dismissing the rest of the information, so I wonder why? Is there value in the rest of the data in the list which we should be using, my guess is there is

Reply:
I'm still figuring out it from Exante.

Reply:
The response from the Exante support:

> In relation to your question, please, let us inform you about the following:
upon requesting an instrument quote stream, you will be receiving a single result for a bid and a single result for an ask.

I have only seen only single element in the array. I suppose this might be for some future functionality.